### ISSUE TITLE
Plugin should have access to its executionId

### ISSUE THREAD

**Issue by brollb** [webgme/webgme-engine#169]
Currently, the `client` has access to a plugin's execution ID but this is not available to the plugin itself. This would be really nice so the executionId could be recorded by the plugin (and enable it to be canceled from a different browser session).

**Comment by kecso** [comment]
I agree that it would be beneficial to have the id stored on the server side and made accessible to the initiating user so other client access would have the potential to communicate with it.
On the other hand, I do not see how allowing the plugin to know its own id would help this situation...

**Comment by brollb** [comment]
I had imagined that the plugin could determine where to store the executionId. In my use-case, the execution is associated with part of the model so I have been imagining that the plugin would storet he executionId with the associated part of the model.

**Comment by brollb** [comment]
Does the use case make sense? Basically, I would like to associate the execution with a part of the model by storing the execution ID in a node attribute.

**Comment by brollb** [comment]
@kecso Would you be open to a PR changing this? Or do you have a better recommendation for how to associate a running plugin to a part of the model?

This is currently causing some annoying forking and related issues...
